cmd/asset-syncer/delete.go

```diff
@@ -33,34 +36,69 @@ var deleteCmd = &cobra.Command{
 			cmd.Help()
 			return
 		}
-		mongoURL, err := cmd.Flags().GetString("mongo-url")
-		if err != nil {
-			logrus.Fatal(err)
-		}
-		mongoDB, err := cmd.Flags().GetString("mongo-database")
-		if err != nil {
-			logrus.Fatal(err)
-		}
-		mongoUser, err := cmd.Flags().GetString("mongo-user")
-		if err != nil {
-			logrus.Fatal(err)
-		}
-		mongoPW := os.Getenv("MONGO_PASSWORD")
 		debug, err := cmd.Flags().GetBool("debug")
 		if err != nil {
 			logrus.Fatal(err)
 		}
 		if debug {
 			logrus.SetLevel(logrus.DebugLevel)
 		}
-		mongoConfig := datastore.Config{URL: mongoURL, Database: mongoDB, Username: mongoUser, Password: mongoPW}
-		dbSession, err := datastore.NewSession(mongoConfig)
-		if err != nil {
-			logrus.Fatalf("Can't connect to mongoDB: %v", err)
+		database, err := cmd.Flags().GetString("database-type")
+		var manager assetManager
+		if database == "mongodb" {
+			mongoURL, err := cmd.Flags().GetString("mongo-url")
+			if err != nil {
+				logrus.Fatal(err)
+			}
+			mongoDB, err := cmd.Flags().GetString("mongo-database")
```

Comment:
either way, we really should not be parsing flags in this file... they should be parsed (and validated) in the main.go to create an instance of your interface, avoiding sprinkling `.Fatal` errors throughout the code.